src/System.Management.Automation/engine/runtime/Operations/MiscOps.cs

```diff
@@ -1047,6 +1081,15 @@ public override string ToString()
         //    dir > out
         internal override void Bind(PipelineProcessor pipelineProcessor, CommandProcessorBase commandProcessor, ExecutionContext context)
         {
+            if (commandProcessor is NativeCommandProcessor nativeCommand
+                && nativeCommand.CommandRuntime.ErrorMergeTo is not MshCommandRuntime.MergeDataStream.Output
+                && FromStream is RedirectionStream.Output
+                && !string.IsNullOrWhiteSpace(File))
+            {
+                nativeCommand.StdOutDestination = FileBytePipe.Create(File, Appending);
```

Comment:
This is currently just opening a file stream. It needs path resolution to be done through `FileSystemProvider` and it needs to surface errors similarly to how `Out-File` does today.